src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addSummaryBlock/index.js

```diff
@@ -0,0 +1,40 @@
+import pathOr from 'ramda/src/pathOr';
+import deepClone from 'ramda/src/clone';
+import path from 'ramda/src/path';
+import { singleTextBlock } from '#app/models/blocks';
+
+const getSummary = json => {
+  const summary = pathOr(null, ['promo', 'summary'], json);
+
+  if (!summary) {
+    return json;
+  }
+
+  return summary;
+};
+
+const addSummaryBlocks = originalJson => {
+  const json = deepClone(originalJson);
+  const pageType = path(['metadata', 'type'], json);
+  const blocks = path(['content', 'model', 'blocks'], json);
+
+  if (pageType !== 'PGL' || !blocks) {
+    return json;
+  }
+
+  const headline = blocks.find(({ type }) =>
+    ['headline', 'fauxHeadline'].includes(type),
+  );
+
+  const summary = singleTextBlock(getSummary(json));
+
+  const remainingBlocks = blocks.filter(({ type }) => {
+    return !['headline', 'fauxHeadline'].includes(type);
+  });
+
+  json.content.model.blocks = [headline, summary, ...remainingBlocks];
```

Comment:
Just to clarify, are we lifting and shifting `headline` here so that `summary` can be positioned accordingly?

Reply:
Does this relate to the discussion above at https://github.com/bbc/simorgh/pull/5153/files#r367517021 or are you making a different point?

Reply:
My point is about mutating the block model, which I think relates to your concern about potential side-effects. All of these things can be addressed in https://github.com/bbc/simorgh/issues/5175.